### PR TITLE
DOC: `sparse_distance_matrix` and `count_neighbors` examples

### PR DESCRIPTION
**Pull request by AtsushiSakai**
#### Reference issue

fix a part of #7168

#### What does this implement/fix?

I added Examples for `sparse_distance_matrix` and `count_neighbors` of `spatial.KDTree` and `spatial.cKDTree`.

### CODE REVIEW

File: scipy/spatial/ckdtree.pyx [hunk 1]
```diff
@@ -1448,6 +1469,26 @@ cdef class cKDTree:
...
+        You can compute a sparse distance matrix between two kd-trees:
+
+        >>> import numpy as np
+        >>> from scipy.spatial import cKDTree
+        >>> np.random.seed(21701)
+        >>> points1 = np.random.random((5, 2))
+        >>> points2 = np.random.random((5, 2))
+        >>> kd_tree1 = cKDTree(points1)
+        >>> kd_tree2 = cKDTree(points2)
+        >>> sdm = kd_tree1.sparse_distance_matrix(kd_tree2, 1.0)
```
Comment (peterbell10) [inline scipy/spatial/ckdtree.pyx:1484]: Perhaps reduce  the radius to 0.2 so we can see the sparsity in the matrix?

Comment (peterbell10) [reply scipy/spatial/ckdtree.pyx:1484]: Maybe even compare to `spatial.distance_matrix` to show how the sparsity matches with the distances above the threshold.

Comment (AtsushiSakai) [reply scipy/spatial/ckdtree.pyx]: good idea. I fixed it. PTAL.

**Review by peterbell10 — APPROVED** [review]
LGTM :+1:

**Comment by peterbell10** [comment]
Thanks @AtsushiSakai

**Pull request merged**